src/components/Carousel.tsx

```diff
@@ -1,18 +1,245 @@
+            );
+          })}
+        </ul>
+
+        <div className="Carousel__buttons">
+          <button
+            disabled={movedDistance === 0 && infinite === false}
+            className="Carousel__button Carousel__button--prev"
+            type="button"
+            onClick={this.handleImagesPrev}
+          >
+            Prev
+          </button>
+
+          <button
+            data-cy="next"
+            disabled={movedDistance >= ((imagesAmount - frameSize)
+              * (itemWidth + gap)) && infinite === false}
+            className="Carousel__button Carousel__button--next"
+            type="button"
+            onClick={this.handleImagesNext}
+          >
+            Next
+          </button>
+        </div>
+
+        <form className="Carousel__form">
+          <label className="Carousel__form__label">
+            {'Moving step: '}
+            <input
+              className="Carousel__form__input"
+              type="number"
+              value={step}
+              min="1"
+              max="5"
+              step="1"
+              onChange={(e) => {
+                this.setState({ step: +e.target.value });
+              }}
+            />
+          </label>
+          <label className="Carousel__form__label">
+            {'Frame size: '}
+            <input
+              className="Carousel__form__input"
+              type="number"
+              value={frameSize}
+              min="2"
+              max="5"
+              step="1"
+              onChange={(e) => {
+                this.setState({ frameSize: +e.target.value });
+              }}
+            />
+          </label>
+          <label className="Carousel__form__label">
+            {'Image size: '}
+            <input
+              className="Carousel__form__input"
+              type="number"
+              value={itemWidth}
+              min="50"
+              max="200"
+              step="10"
+              onChange={(e) => {
+                this.setState({ itemWidth: +e.target.value });
+              }}
+            />
+          </label>
+          <label className="Carousel__form__label">
+            {'Motion speed: '}
+            <input
+              className="Carousel__form__input"
+              type="number"
+              value={animationDuration}
+              min="300"
+              max="3000"
+              step="100"
+              onChange={(e) => {
+                this.setState({ animationDuration: +e.target.value });
+              }}
+            />
+          </label>
+          <label className="Carousel__form__label">
+            {'Infinite: '}
+            <input
+              className="Carousel__form__input Carousel__form__input--checkbox"
+              type="checkbox"
+              onClick={() => {
+                if (infinite === false) {
+                  this.setState({ infinite: true });
+                } else {
+                  this.setState({ infinite: false });
+                }
```

Comment:
```suggestion
              
                  this.setState((prevState) => { ...prevState, infinite: !prevState.infinite });
         
```
you can do something like this, or try to set  `event.currentTarget.checked` value